src/libstd/memchr.rs

```diff
@@ -0,0 +1,339 @@
+/* Initial implementation taken from Andrew Gallant's rust-memchr
```

Comment:
Needs a real copyright header, see other files.

If you put in burntsushi's name, I want to be credited too -- the pure rust memchr impls are mine.